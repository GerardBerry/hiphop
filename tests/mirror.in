;
;
I3(lala);
!eval-begin;
example.prg.set_input("I4", 34);
example.prg.set_input("I1")
example.prg.set_input("I3", "hihihiihi");
!eval-end;
;
I4(5);
I4;
