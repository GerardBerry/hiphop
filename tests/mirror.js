"use hopscript"

var hh = require("hiphop");

function foo(evt) {
   console.log("hi from foo signal", evt.signalName, "is set!")
}

function bar(evt) {
   console.log("hi from bar signal", evt.signalName, "is set with value",
	       evt.signalValue, "!")
}

var prg = <hh.module>
  <hh.inputsignal name="I1" />
  <hh.outputsignal name="O1"/>
  <hh.inputsignal name="I2" valued/>
  <hh.outputsignal name="O2" valued/>
  <hh.inputsignal name="I3" valued/>
  <hh.outputsignal name="O3" valued/>
  <hh.inputsignal name="I4" valued />
  <hh.outputsignal name="O4" valued/>
  <hh.loop>
    <hh.sequence>
      <hh.present signal_name="I1">
	<hh.emit signal_name="O1"/>
      </hh.present>
      <hh.present signal_name="I2">
	<hh.emit signal_name="O2" args=${hh.value("I2")}/>
      </hh.present>
      <hh.present signal_name="I3">
	<hh.emit signal_name="O3" args=${hh.value("I3")}/>
      </hh.present>
      <hh.present signal_name="I4">
	<hh.emit signal_name="O4" args=${hh.value("I4")}/>
      </hh.present>
      <hh.pause/>
    </hh.sequence>
  </hh.loop>
</hh.module>

var m = new hh.ReactiveMachine(prg, "mirror");
m.addEventListener("O1", foo);
m.addEventListener("O2", bar);
m.addEventListener("O3", bar);
m.addEventListener("O4", bar);

exports.prg = m
