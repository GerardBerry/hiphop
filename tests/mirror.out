mirror> ;
--- Output:
mirror> ;
--- Output:
mirror> I3;
--- Output: O3(lala)
hi from bar signal O3 is set with value lala !
mirror> I1 I3 I4;
--- Output: O1 O3(hihihiihi) O4(34)
hi from foo signal O1 is set!
hi from bar signal O3 is set with value hihihiihi !
hi from bar signal O4 is set with value 34 !
mirror> I4;
--- Output: O4(5)
hi from bar signal O4 is set with value 5 !
mirror> I4;
--- Output: O4(5)
hi from bar signal O4 is set with value 5 !
